app/javascript/components/region-form/index.jsx

```diff
@@ -0,0 +1,71 @@
+import React, {Component, useState} from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import createSchema from './region-form.schema';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import {API} from '../../http_api';
+import handleFailure from "../../helpers/handle-failure";
+import miqRedirectBack from "../../helpers/miq-redirect-back";
+
+class RegionForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      schema: createSchema(props.maxDescLen),
+    };
+  }
+
+  componentDidMount() {
+    miqSparkleOn();
+    API.get(`/api/regions/${this.props.regionFormId}`)
+      .then((data) => this.setState({ initialValues: { ...data } }))
+      .then(miqSparkleOff)
+      .catch(handleFailure);
+  }
+
+  onCancel = () => {
+    const message = sprintf(__('Edit of Region was cancelled by the user'));
+    miqRedirectBack(message, 'success', `/ops/explorer/?button=cancel`);
+  };
+
+  onSubmit = (values) => {
+    miqSparkleOn();
+    const message = sprintf(__('Region was saved'));
+    const { regionFormId } = this.props;
+    const submitUrl = `/ops/explorer/${regionFormId}?button=save`;
+    const request = () => API.post(`/api/regions/${regionFormId}`, {
+      action: 'edit',
+      resource: {
+        description: values.description,
+      }
+    });
+    return request().then(() => miqRedirectBack(message, 'success', submitUrl)).catch(miqSparkleOff);
+  }
+
+  render() {
+    return (
+      <Grid fluid style={{ paddingTop: 20 }}>
+        <MiqFormRenderer
+          initialValues={this.state.initialValues}
+          schema={this.state.schema}
+          onSubmit={this.onSubmit}
+          onCancel={this.onCancel}
+          onReset={() => add_flash(__('All changes have been reset'), 'warn')}
+          canReset
+          buttonsLabels={{
+            submitLabel: __('Save'),
+            resetLabel: __('Reset'),
+            cancelLabel: __('Cancel'),
+          }}
+        />
+      </Grid>
+    );
+  }
+}
+
+RegionForm.propTypes = {
+  maxDescLen: PropTypes.number.isRequired,
+  regionFormId: PropTypes.number.isRequired,
```

Comment:
If this is just an `id` field, why not name it `id` or `regionId` if we want to be precise?